src/main/node/gatsby.ts

```diff
@@ -51,3 +51,30 @@ export function installGatsby(): Promise<void> {
     });
   });
 }
+
+/**
+ * Generate new PaaW project with gatsby.
+ * Rejects the promise if a command execution error occurs.
+ * @return {Promise<void>}
+ */
+export function generateProject(
+  dir: string,
+  projectName: string,
+  templateUrl: string
+): Promise<void> {
+  return new Promise<void>((resolve, reject) => {
+    appendLog('Generating new project...');
+    exec(
+      `gatsby new ${projectName} ${templateUrl}`,
+      { cwd: dir },
```

Comment:
Added the `cwd` (short for current working directory) in order to generate the project in the default directory specified by the user for the app. Otherwise the project would generate in the root of the app.